apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/ExtensionContext.java

```diff
@@ -69,14 +65,12 @@ void deserialize(String value) {
         final String[] extensionParts = value.split("-");
         // All parts of the extension header are optional.
         // only try to read it when it exist.
-        if (extensionParts.length > 0) {
+        if (extensionParts.length == 2) {
```

Comment:
Again, don't use `2`.

Reply:
ok

Reply:
Because all parameters are passing rather than optional fields, the length of extensionParts is fixed.

now, the passing data maybe "0-" or "0-1602743904804", length of extensionParts must be 2.
in the feature, if add another field, the propagate data may be "0--" or "0-xxx-xxx". And the length of extensionParts must be 3.

I think logic greater than 0 is redundant

Reply:
Don't require `must`, that is the key. If the new agent propagates `0-xxx-xxx`, I hope the older agent still works, that is why I wrote this part as `>0`, and ask you to do `>1`.

Reply:
I finally understand the key: the compatibility.

Reply:
Yes, this kind of PR shows me the case is, the extension header may be used more in the future.